### ISSUE TITLE
Can't make cxx:FileHeader accept doxygen-style file headers

### ISSUE THREAD

**Issue by deadbird99** [SonarOpenCommunity/sonar-cxx#1787]
Hi.

I've been struggling for two days, trying to make sonar-cxx accept my doxygen styled file headers. They are styled like so:

```cpp
/**
 * \file name of the file
 * \author name <name@domain.com>
 * \copyright copyright text that spans
 *            on multiple lines
 * \brief The description of this file.
 */
```

Si far, I unsuccessfully tested these regexes:

```regex
\/\*\*\n \* \\file [[:print:]]*\n \* \\author [[:print:]]*\n \* \\copyright [[:print:]]*\n \* \\brief [[:print:]]*\n \*\/
\/\*\*\n \* \\file (.*)\n \* \\author (.*)\n \* \\copyright (.*)\n \* \\brief (.*)\n \*\/
\/\*\*\n \* \\file (.+)\n \* \\author (.+)\n \* \\copyright (.|\n)+\\brief(.+)+\n \*\/
\/\*\*\n \* \\file (.+)\n \* \\author (.+)\n \* \\copyright (.)+\\brief(.+)+\n \*\/
```

I even tested a very permissive one

```regex
\/\*\*\n(.+)\n*\/
```

Some of them had `/regex/s` flags, just to try. I tested all of these using [Regex101](https://regex101.com/), they all work fine. But when inserted in `cxx:FileHeader` rule, they either crash the whole analysis, or just simply trigger false positives.

Right now, I'm in a dead end, none of these worked. So I deactivated the rule, but it's clearly not the way I want it to be.

My sonar-cxx version: 1.3.1
My SonarQube version: 7.9.1.

Thanks a lot for the help.

**Comment by guwirth** [comment]
Hi @deadbird99,

> struggling for two days, trying to make sonar-cxx accept my doxygen styled file headers

What exactly are you trying? Which problem do you like to solve?

> cxx:FileHeader

Detect a file header ? Maybe it helps if you are looking in the source code?
https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/cxx-checks/src/main/java/org/sonar/cxx/checks/regex/FileHeaderCheck.java

Regards,

**Comment by deadbird99** [comment]
Hi @guwirth , and thanks for the prompt answer.

Here's my workflow. I'm working on a C++ project, and I document it with Doxygen. Doxygen comments I use are formatted the way I mentionned above, and sonar-cxx triggers a cxx::FileHeader rule whenever ths code analysis is run, telling me that the file has no header comment.
So I tried to modify cxx:FileHeader's regex (see above), but no luck.

**Comment by guwirth** [comment]
Please have a look to the unit tests and sample files, maybe this helps.

Unit tests:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/cxx-checks/src/test/java/org/sonar/cxx/checks/regex/FileHeaderCheckTest.java

Samples:
https://github.com/SonarOpenCommunity/sonar-cxx/tree/master/cxx-checks/src/test/resources/checks/FileHeaderCheck

**Comment by deadbird99** [comment]
Thanks a lot. I'm closing the issue for now.
